cfe_internal/update/windows_unattended_upgrade.cf

```diff
@@ -0,0 +1,75 @@
+bundle agent windows_unattended_upgrade
+{
+  vars:
+
+    windows::
+
+      "_schedule_upgrade_script"
+        string => "$(const.dirsep)schedule_unattended_upgrade.ps1",
+        comment => "sys.bindir has issues when used as execution path"; # TODO examine and fix/document specific issue(s)
+
+      "_upgrade_script"
+        string => "$(sys.bindir)$(const.dirsep)unattended-upgrade.bat";
+
+      "_min_offset" string => "2";
+
+  files:
+    windows::
+      # /sigh/. In order to make a stable promise, we render a script that runs
+      # to create the scheduled task that runs another script to perform the
+      # upgrade outside of cf-agent. Since the command differs based on the time
+      # of of day, the only way to have a stable promise is to promise a script
+      # where the time base difference is handeled inside the script. One
+      # alternative would be to use inline subshell exection (iex)
+      "$(_schedule_upgrade_script)"
+        create => "true",
+        edit_template_string => '
+$scheduled_time_to_upgrade = (Get-Date).AddMinutes($(_min_offset)).ToString("HH:mm")
+$(sys.winsysdir)$(const.dirsep)schtasks.exe /Create /SC ONCE /TN "CFEngine Agent Upgrade" /TR "$(_upgrade_script)" /ST $scheduled_time_to_upgrade /RU "System" /F
+',
+        template_method => "inline_mustache",
+        template_data => '{}';
+
+      "$(_upgrade_script)"
+        edit_template_string => `
+REM kill all cf procs first so that installed can run
+Taskkill /IM cf-execd.exe /F
+Taskkill /IM cf-monitord.exe /F
+Taskkill /IM cf-serverd.exe /F
+Taskkill /IM cf-agent.exe /F
+echo File=$(cfengine_software.local_software_dir)$(const.dirsep)$(cfengine_package_names.my_pkg) | "$(sys.workdir)$(const.dirsep)modules$(const.dirsep)packages$(const.dirsep)msiexec.bat" file-install
+REM exit 0 so that scheduled task will not keep trying a failing situation
+REM looking at package module logs of one failure should help in debugging
+exit 0
+`,
+        template_method => "inline_mustache",
+        template_data => '{}',
+        create => "true",
+        if => not( strcmp( $(sys.cf_version), $(cfengine_software.pkg_version) ));
+
+      "$(_upgrade_script)"
```

Comment:
@nickanderson can you think of a reason why we would not also cleanup `"$(_schedule_upgrade_script)"`?

Reply:
Well, maybe cleaning it up would be considered unnecessary churn.

Reply:
right, I was suspecting as much. I'll remove/refactor back to leaving it.

Reply:
we chatted, it should be fine to remove, I'll test one more time and merge.